src/providers/ipa/ipa_selinux.c

```diff
@@ -712,21 +702,19 @@ static errno_t selinux_fork_child(struct selinux_child_state *state)
     pid = fork();
 
     if (pid == 0) { /* child */
-        exec_child_ex(state, pipefd_to_child, pipefd_from_child,
+        exec_child_ex(state, pipefd_to_child, NULL,
                       SELINUX_CHILD, SELINUX_CHILD_LOG_FILE, extra_args,
                       false, STDIN_FILENO, STDOUT_FILENO);
         DEBUG(SSSDBG_CRIT_FAILURE, "Could not exec selinux_child: [%d][%s].\n",
               ret, sss_strerror(ret));
         return ret;
     } else if (pid > 0) { /* parent */
-        state->io->read_from_child_fd = pipefd_from_child[0];
-        close(pipefd_from_child[1]);
+        state->io->read_from_child_fd = -1;
         state->io->write_to_child_fd = pipefd_to_child[1];
         close(pipefd_to_child[0]);
-        sss_fd_nonblocking(state->io->read_from_child_fd);
         sss_fd_nonblocking(state->io->write_to_child_fd);
 
-        ret = child_handler_setup(state->ev, pid, NULL, NULL, NULL);
+        ret = child_handler_setup(state->ev, pid, selinux_child_done, req, NULL);
```

Comment:
Hi,

I think you should save the last argument `child_ctx` into the state so that in case the writing to the child failed you can call `child_handler_destroy()` in `selinux_child_step()` (or the new name) to make sure the child is not lingering around.

bye,
Sumit